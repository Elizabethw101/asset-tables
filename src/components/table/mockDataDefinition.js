
 export const mockDataDefinition = [
        {
          label: "Ticker",
          dataType: "text",
          sortType: "alpha",
          dataField: "ticker"
        },
        {
          label: "Price",
          dataType: "currency",
          sortType: "number",
          dataField: "price"
        },
        {
          label: "Asset Class",
          dataType: "asset",
          sortType: "assetClass",
          dataField: "assetClass"
        }
      ];
