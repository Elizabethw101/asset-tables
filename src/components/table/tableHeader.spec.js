/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import TableHeader from "./tableHeader";
import { mockDataDefinition } from "./mockDataDefinition";
import { act } from "react-dom/test-utils";

describe("it tests the table header component", () => {
  let container = null;
  let resetData = jest.fn();
  let sortTableData = jest.fn();

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    container = null;
  });

  it("loads and renders the table header", () => {
    container = renderer.create(
      <TableHeader
        resetData={resetData}
        headerData={mockDataDefinition}
        sortData={sortTableData}
      />
    );
    expect(container.toJSON()).toMatchSnapshot();
  });

  it("should have default props", () => {
    expect(TableHeader.defaultProps.sortData).toBeDefined();
  });

  it("it expects the sort function to be called on clicking Ticker", () => {
    const container = shallow(
      <TableHeader
        resetData={resetData}
        headerData={mockDataDefinition}
        sortData={sortTableData}
      />
    );

    const firstHeader = container.find("th").first();
    firstHeader.simulate("click");
    expect(sortTableData).toHaveBeenCalledTimes(1);
  });

  it("it expects the reset sort to be called on two clicks", () => {
    const container = mount(
      <table>
        <TableHeader
          headerData={mockDataDefinition}
          sortData={sortTableData}
        />
      </table>
    );
    const firstHeader = container.find("th").first();
    firstHeader.simulate("click");
    expect(firstHeader.getDOMNode()).toHaveClass('sort')
    firstHeader.simulate("click");
    expect(firstHeader.getDOMNode()).not.toHaveClass('sort')
    expect(sortTableData).toHaveBeenCalled();
  });
});
