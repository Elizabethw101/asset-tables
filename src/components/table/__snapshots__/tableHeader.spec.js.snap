// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it tests the table header component loads and renders the table header 1`] = `
<thead>
  <tr
    className="header"
  >
    <th
      className="hcell"
      onClick={[Function]}
    >
      Ticker
    </th>
    <th
      className="hcell"
      onClick={[Function]}
    >
      Price
    </th>
    <th
      className="hcell"
      onClick={[Function]}
    >
      Asset Class
    </th>
  </tr>
</thead>
`;
